src/lib/optimizer/enumerate_ccp.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <cstdint>
+#include <optional>
+#include <utility>
+#include <vector>
+
+#include "join_graph_edge.hpp"
+
+namespace opossum {
+
+class JoinGraph;
+
+/**
+ * CsgCmpPair enumeration algorithm described in in https://dl.acm.org/citation.cfm?id=1164207
+ */
+class EnumerateCcp final {
```

Comment:
General remark: Without reading it is quite hard to get what you are doing in this class. Could you add a short comment explaining the most basic steps?

Reply:
Also `in in`.

And please add in quotes what paper you are referring to. This is not a DOI link which should be more stable over years. ACM is currently restructuring the DL site. I hope that won't screw up existing links but you never know.